hail_search/hail_search_query.py

```diff
@@ -521,11 +516,41 @@ def _filter_by_in_silico(self, in_silico_filters):
                 missing_q &= q
             in_silico_qs.append(missing_q)
 
-        in_silico_q = in_silico_qs[0]
-        for q in in_silico_qs[1:]:
-            in_silico_q |= q
+        self._ht = self._ht.filter(self._or_filter(in_silico_qs))
+
+    def _get_in_silico_filter(self, in_silico, value):
+        score_path = self.PREDICTION_FIELDS_CONFIG[in_silico]
+        enum_lookup = self._get_enum_lookup(*score_path)
+        if enum_lookup is not None:
+            ht_value = self._ht[score_path.source][f'{score_path.field}_id']
+            score_filter = ht_value == enum_lookup[value]
+        else:
+            ht_value = self._ht[score_path.source][score_path.field]
+            score_filter = ht_value >= float(value)
+
+        return score_filter, ht_value
+
+    def _filter_by_annotations(self, pathogenicity, annotations):
+        annotation_override_filters = self._get_annotation_override_filters(pathogenicity, annotations)
+
+        self._annotate_allowed_consequences(annotations, annotation_override_filters)
+        if 'has_allowed_annotation' in self._ht.row:
```

Comment:
Using `hasattr(self._ht, 'has_allowed_annotation')` is more straightforward.

Reply:
I prefer it this way, as it is specifically requiring it to be a row annotation, and not a table function or a global